### PR TITLE
[MMSYS] Minor fixes for FreeSoundFiles

### PR DESCRIPTION
**Pull request by TAN-Gaming**
## Purpose
- Addendum to 781c247bd3c5aeee799527db2ce06e15f794b0a4

## Proposed changes
- Skip empty (None) item
- Remove unnecessary `pSoundPath` variable

### CODE REVIEW

**Comment by learn-more** [comment]
Does this fix a bug?
What is the purpose of this PR?
Please describe this in your description.

**Comment by TAN-Gaming** [comment]
> Does this fix a bug? What is the purpose of this PR? Please describe this in your description.

I tried debugging (using text output). `free()` is called on an empty item (None).

It doesn't make sense to free an empty item (None) so just skip it.

File: dll/cpl/mmsys/sounds.c [hunk 1]
```diff
@@ -886,12 +886,12 @@ FreeSoundFiles(HWND hwndDlg)
...
         lResult = ComboBox_GetItemData(hwndComboBox, lIndex);
-        if (lResult == CB_ERR)
+        if (lResult == CB_ERR || lResult == 0)
         {
             continue;
         }
 
-        pSoundPath = (PWCHAR)lResult;
+        pSoundPath = (LPWSTR)lResult;
         free(pSoundPath);
```
Comment (HBelusca) [inline dll/cpl/mmsys/sounds.c:895]: Could have been directly: `free((PVOID)lResult)`;

Comment (TAN-Gaming) [reply dll/cpl/mmsys/sounds.c]: Done.

**Comment by learn-more** [comment]
> If ptr is a null pointer, the function does nothing.

https://en.cppreference.com/w/c/memory/free
So not really fixing a bug, hence my question.

**Comment by TAN-Gaming** [comment]
> > If ptr is a null pointer, the function does nothing.
> 
> https://en.cppreference.com/w/c/memory/free So not really fixing a bug, hence my question.

It won't fix any bug, but the purpose of the function is to free all the allocated items.
This change doesn't introduce more `if`.
The `lResult == 0` check is widely used too.

File: dll/cpl/mmsys/sounds.c [hunk 2]
```diff
@@ -886,13 +885,12 @@ FreeSoundFiles(HWND hwndDlg)
...
         lResult = ComboBox_GetItemData(hwndComboBox, lIndex);
-        if (lResult == CB_ERR)
+        if (lResult == CB_ERR || lResult == 0)
         {
             continue;
         }
 
-        pSoundPath = (PWCHAR)lResult;
-        free(pSoundPath);
+        free((PVOID)lResult);
```
Comment (binarymaster) [inline dll/cpl/mmsys/sounds.c:893]: I would prefer the former code, it's more readable and self-descriptive.

**Comment by TAN-Gaming** [comment]
Waiting for reviews ...

**Comment by learn-more** [comment]
Sorry for the long wait.
Considering @binarymaster prefers the previous code, and I am not convinced either that this is an improvement, let's close it.

**Pull request closed without merging**